static/app/views/projectInstall/createProject.tsx

```diff
@@ -112,107 +206,14 @@ function CreateProject() {
-          : `/organizations/${slug}/experimental/projects/`;
-        const projectData = await api.requestPromise(url, {
-          method: 'POST',
-          data: {
-            name: projectName,
-            platform: selectedPlatform.key,
-            default_rules: defaultRules ?? true,
-            origin: 'ui',
-          },
-        });
-
-        const ruleIds: string[] = [];
-        if (shouldCreateCustomRule) {
-          const ruleData = await api.requestPromise(
-            `/projects/${organization.slug}/${projectData.slug}/rules/`,
-            {
-              method: 'POST',
-              data: {
-                name,
-                conditions,
-                actions,
-                actionMatch,
-                frequency,
-              },
-            }
-          );
-          ruleIds.push(ruleData.id);
-        }
-        const ruleData = await createNotificationAction({
-          shouldCreateRule,
-          name,
-          projectSlug: projectData.slug,
-          conditions,
-          actionMatch,
-          frequency,
-        });
-        if (ruleData) {
-          ruleIds.push(ruleData.id);
-        }
-        trackAnalytics('project_creation_page.created', {
-          organization,
-          issue_alert: defaultRules
-            ? 'Default'
-            : shouldCreateCustomRule
-              ? 'Custom'
-              : 'No Rule',
-          project_id: projectData.id,
-          platform: selectedPlatform.key,
-          rule_ids: ruleIds,
-        });
-
-        ProjectsStore.onCreateSuccess(projectData, organization.slug);
-
-        if (team) {
-          addSuccessMessage(t('Created project %s', `${projectData.slug}`));
-        } else {
-          addSuccessMessage(
-            t(
-              'Created %s under new team %s',
-              `${projectData.slug}`,
-              `#${projectData.team_slug}`
-            )
-          );
-        }
-
-        browserHistory.push(
-          normalizeUrl(
-            makeProjectsPathname({
-              orgSlug: organization.slug,
-              path: `/${projectData.slug}/getting-started/`,
-            })
-          )
-        );
-      } catch (err) {
-        setInFlight(false);
-        setErrors(err.responseJSON);
-        addErrorMessage(t('Failed to create project %s', `${projectName}`));
-
-        // Only log this if the error is something other than:
-        // * The user not having access to create a project, or,
-        // * A project with that slug already exists
-        if (err.status !== 403 && err.status !== 409) {
-          Sentry.withScope(scope => {
-            scope.setExtra('err', err);
-            Sentry.captureMessage('Project creation failed');
-          });
-        }
-      }
+      createProject.mutate({
+        name: projectName,
+        platform: selectedPlatform,
+        default_rules: alertRuleConfig?.defaultRules ?? true,
+        firstTeamSlug: team,
+      });
```

Comment:
imho would be cleaner to use mutateAsync here and have all the `onError` and `onSuccess` handling in this function here.
e.g. updating `inFlight` will no be split over multiple functions anymore.

```suggestion
      await createProject.mutateAsync({
        name: projectName,
        platform: selectedPlatform,
        default_rules: alertRuleConfig?.defaultRules ?? true,
        firstTeamSlug: team,
      });
```

Reply:
never used this before. thanks for the suggestion!